src/data/real/irrational.lean

```diff
@@ -257,4 +257,52 @@ open irrational
 @[simp] theorem irrational_inv_iff : irrational x⁻¹ ↔ irrational x :=
 ⟨of_inv, irrational.inv⟩
 
+/-!
+### Lemmas about Irrationals between Reals
+-/
+
+/-- An irrational number between any two distinct rational numbers.-/
+noncomputable def irrational_btwn_rats {x y : ℚ} (h : x < y) : { r : ℝ // irrational r } :=
+⟨x +((y - x) : ℚ) * (real.sqrt 2)⁻¹, begin
+  refine irrational.rat_add _
+    (irrational.rat_mul
+      (irrational_inv_iff.mpr irrational_sqrt_two) _),
+  exact_mod_cast (ne_of_gt (sub_pos_of_lt h)) end⟩
```

Comment:
If you want this can be a mostly-term-mode proof: instead of using `refine`, just replace the relevant `_` with `by exact_mod_cast ...`. Completely optional, of course.